build.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/bash
+
+set -e  # exit whenever command in pipeline fails
+set -x  # print commands as they are executed
+
+VERSION=`grep "Version:.*[0-9]" slurm.spec | tr -s " " |  awk '{print $2;}'`
+RELEASE=`grep "%define rel.*[-1-9]" slurm.spec | tr -s " " | awk '{print $3}'`
+
+if [ "${RELEASE}" != "1" ]; then
+    SUFFIX=${VERSION}-${RELEASE}
+else
+    SUFFIX=${VERSION}
+fi
+
+GITTAG=$(git log --format=%ct.%h -1)
+
+SCRIPT=$(readlink -f "${BASH_SOURCE[0]}")
+ORIGIN=$(dirname "$SCRIPT")
+
+# which version to download from github
+SLURM_VERSION=${VERSION:-24.05.3}
+UPSTREAM_REL=${UPSTREAM_REL:-1}
+
+# which release should be used for our RPMs
+OUR_RELEASE=${RELEASE:-1}
+
+# NVML
+# allow _empty_ version, which is used in pipeline
+
+if grep "release 8.8" /etc/redhat-release; then
+   NVIDIA_DRIVER=${NVIDIA_DRIVER-550.90.07}
+   NVDRV_NVML_PKG="nvidia-driver-NVML${NVIDIA_DRIVER:+-$NVIDIA_DRIVER}"
+   CUDA_VERSION=${CUDA_VERSION:-12.6}
+   CUDA_NVML_PKG="cuda-nvml-devel-${CUDA_VERSION//./-}"
+elif grep "release 9.4" /etc/redhat-release; then
+    NVDRV_NVML_PKG="libnvidia-ml"
+    CUDA_VERSION=${CUDA_VERSION:-12.6}
+    CUDA_NVML_PKG="cuda-nvml-devel-${CUDA_VERSION//./-}"
+fi
+
+# Prepare directory structure
+rm -Rf $ORIGIN/rpmbuild/ $ORIGIN/dist/
+mkdir -p $ORIGIN/rpmbuild/{BUILD,RPMS,SRPMS,SOURCES} $ORIGIN/dist
+echo "Building source tarball"
+
+# archive git repo
+echo "archive suffix: ${SUFFIX}"
+git archive --format=tar.gz -o "rpmbuild/SOURCES/slurm-${SUFFIX}.tar.gz" --prefix="slurm-${SUFFIX}/" HEAD
+
+cp slurm.spec "SPECS"
+
+# Patch sources
+#for src_patch in $ORIGIN/src-patches/*.patch; do
+    #echo "Patching $src_patch"
+    #git am $src_patch
+#done
+
+# Patch spec file
+#for spec_patch in $ORIGIN/spec-patches/*.patch; do
+    #echo "Patching $spec_patch"
+    #patch -p1 -b -i $spec_patch
+#done
+
+# Install dependencies
+# see https://slurm.schedmd.com/quickstart_admin.html#build_install
+
+echo "Installing specfile requires"
+# this goes first because it might install undesired stuff related to `--with` options
+sudo dnf -y builddep slurm.spec
+
+# dependecy versions
+
+if grep "release 8.8" /etc/redhat-release; then
+    UCX_VERSION="1.13.1-2.el8.x86_64"
+    PMIX_VERSION=">= 4.2.6"
+    HWLOC_VERSION=">= 2.2.0-3"
+elif grep "release 9.2" /etc/redhat-release; then
```

Comment:
you can remove this

Reply:
Not sure why this is here, it's not in the file on my laptop, so I cannot remove it even if I wanted to.

Reply:
then what branch/commit are you working on?